app/controllers/events_controller.rb

```diff
@@ -0,0 +1,89 @@
+class EventsController < ApplicationController
+  before_filter :authenticate_user!, except: [:index, :show]
+  after_action :verify_authorized
+  before_action :set_user
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+
+  def index
+    if params[:type] == 'past'
+      @events = @user.past_events
+    elsif params[:type] == 'upcoming'
+      @events = @user.upcoming_events
+    else
+      @events = @user.today_events
+    end
+    
+    is_public_only = current_user != @user
+    @events = @events.public_only if is_public_only
+
+    authorize @events
+  end
+
+  def new
+    @event = Event.new(organizer: @user)
+    @event.event_address = EventAddress.new
+    authorize @event
+  end
+
+  def create
+    @event = Event.new(event_params)
+    @event.scheduled_at = parse_scheduled_at
+    authorize @event
+
+    respond_to do |format|
+      if @event.save
+        format.html { redirect_to [@user, @event], notice: 'Event was successfully created.' }
+        format.json { render :show, status: :created, location: @event }
```

Comment:
Does this line with `render :show` and `format.json ...` work ok? I'm wondering if it might be a problem as there isn't a corresponding `views/events/show` template (similarly also see line 51 in the `update` action).